apiserver/facades/controller/caasapplicationprovisioner/provisioner.go

```diff
@@ -704,7 +704,7 @@ func (a *API) ApplicationOCIResources(args params.Entities) (params.CAASApplicat
 			if v.Type != charmresource.TypeContainerImage {
 				continue
 			}
-			reader, err := resources.OpenResource(v.Name)
+			reader, err := resources.OpenResource(v.Name, false)
```

Comment:
This doesn't feel right. It seems to me that OpenResource should always be false, and SetResource should mostly (always) be true.

If you are just *reading* the resource, you aren't changing it, thus I'm not sure we need a way to supply it. My quick grep says you always passed false, but maybe I missed a case. That would help keep the api clean.

Reply:
+1, this is called during the provisioning step in `(a *appWorker) alive()` so I think the resources should already have been set up. And Open should always be read only.

Reply:
Inside the OpenResource we do `set` the resource. 
But I moved this toggle into the `resources.OpenResource` where it calls `set`.